test_six.py

```diff
@@ -875,6 +875,24 @@ class MySlotsWeakref(object):
     assert type(MySlotsWeakref) is Meta
 
 
+@py.test.mark.skipif("sys.version_info[0] < 3")
+def test_add_metaclass_nested():
+    # Regression test for https://github.com/benjaminp/six/issues/259
+    class Meta(type):
+        pass
+
+    class A:
+        class B: pass
+
+    assert A.B.__qualname__.endswith('A.B')
+
+    class A:
+        @six.add_metaclass(Meta)
+        class B: pass
+
+    assert A.B.__qualname__.endswith('A.B')
```

Comment:
I could actually go ahead and test it exactly.  It's just slightly long in this case since it's a class defined inline in a function, so it ends up being `test_add_metaclass_nested.<locals>.A.B` (which is still correct, at least).

Alternatively I could define the test class outside the function.